src/v/cluster/archival_metadata_stm.cc

```diff
@@ -1093,6 +1110,16 @@ void archival_metadata_stm::apply_spillover(const spillover_cmd& so) {
     }
 }
 
+void archival_metadata_stm::apply_replace_manifest(iobuf val) {
+    _manifest->from_iobuf(std::move(val));
+
+    vlog(
+      _logger.debug,
+      "Replace command applied, new start offset: {}, new last offset: {}",
+      get_start_offset(),
+      get_last_offset());
+}
```

Comment:
nit: not for this PR, something I noticed with using the admin reset endpoint, it generally it'd be useful if we had a `partition_manifest::to_short_string()` or somesuch that printed more fields of the manifest but omitted most or all segments (maybe just the first and last?)